File: core/src/main/java/com/starshootercity/OriginsRebornPlaceholderExpansion.java
package com.starshootercity;

import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class OriginsRebornPlaceholderExpansion extends PlaceholderExpansion {
    @Override
    public @NotNull String getIdentifier() {
        return "origin";
    }

    @Override
    public @NotNull String getAuthor() {
        return "cometcake575";
    }

    @Override
    public @NotNull String getVersion() {
        return OriginsReborn.getInstance().getConfig().getString("config-version", "unknown");
    }

    @Override
    public @Nullable String onPlaceholderRequest(Player player, @NotNull String params) {
        Origin origin = OriginSwapper.getOrigin(player);
        if (origin == null) return "";
        return origin.getName();
    }
}

File: core/src/main/java/com/starshootercity/OriginsRebornPlaceholderExpansion.kt
package com.starshootercity

import com.starshootercity.OriginsReborn.Companion.instance
import kotlinx.coroutines.runBlocking
import me.clip.placeholderapi.expansion.PlaceholderExpansion
import org.bukkit.entity.Player

class OriginsRebornPlaceholderExpansion(private val layer: String) : PlaceholderExpansion() {
    override fun getIdentifier(): String {
        return layer
    }

    override fun getAuthor(): String {
        return "cometcake575"
    }

    override fun getVersion(): String {
        return instance.getConfig().getString("config-version", "unknown")!!
    }

    override fun onPlaceholderRequest(player: Player, params: String): String? {
        val origin = runBlocking { OriginSwapper.getOrigin(player, layer) }
        if (origin == null) return ""
        return origin.getName()
    }
}
